### PR TITLE
Serialization

### PR DESCRIPTION
**Pull request by loewenheim**

### CODE REVIEW

File: core/src/main/scala/gapt/formats/json/ProofCollection.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import io.circe.KeyEncoder._
+import cats.implicits._
+
+/**
+ * Wraps a map from proofs to integers. Each proof must be mapped to a
+ * higher number than its subproofs.
+ */
+private[json] class ProofCollection[P <: DagProof[P]] private ( val proofMap: Map[P, Int] ) extends AnyVal
+
+object ProofCollection {
```
Comment (jannikvierling) [inline core/src/main/scala/gapt/formats/json/ProofCollection.scala]: As far as I can tell this object should also have visibility `private[json]`.

File: core/src/main/scala/gapt/formats/json/ProofCollection.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    require( m.forall {
+      case ( p, i ) =>
+        p.immediateSubProofs.forall( q => m.contains( q ) && m( q ) < i )
+    } )
+
+    new ProofCollection( m )
+  }
+}
+
+object ProofCollectionCodec {
```
Comment (jannikvierling) [inline core/src/main/scala/gapt/formats/json/ProofCollection.scala]: This object should also have visibility `private[json]`.

File: core/src/main/scala/gapt/formats/json/ProofCollection.scala [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  /**
+   * Encodes a proof collection
+   */
+  private[json] def proofCollectionEncoder[P <: DagProof[P]]( encodeProof: Encoder[P] => Encoder[P] ): Encoder[ProofCollection[P]] = { coll =>
```
Comment (jannikvierling) [inline core/src/main/scala/gapt/formats/json/ProofCollection.scala]: These lines are a bit too long. Line length should not exceed 120 characters.

File: core/src/main/scala/gapt/formats/json/et/ExpansionTreeCodec.scala [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package gapt.formats.json.et
+import gapt.expr.{ Atom, Expr, Polarity }
+import io.circe.{ Decoder, Encoder }
+import io.circe.syntax._
+//import io.circe.generic.extras.auto._
```
Comment (jannikvierling) [inline core/src/main/scala/gapt/formats/json/et/ExpansionTreeCodec.scala]: This commented out import statement should be removed.

**Comment by loewenheim** [comment]
Thank you for your comments, @jannikvierling. The problems are fixed in https://github.com/gapt/gapt/pull/715/commits/39b40b21b1283977abcd20698905e1e882764731

**Pull request merged**